crates/data-generator/src/blocks/model.rs

```diff
@@ -0,0 +1,105 @@
+use indexmap::IndexMap;
+
+use crate::name::Name;
+
+#[derive(Debug, Eq, PartialEq, Hash, Clone)]
+pub struct Id(pub u32);
+
+#[derive(Debug, Clone)]
+pub struct Block {
+    pub properties: IndexMap<Name, PropertyVariants>,
+    pub states: States,
+}
+
+#[derive(Debug, Clone)]
+pub struct States {
+    pub states: IndexMap<Id, State>,
+    pub default: Id,
+}
+
+impl States {
+    pub fn default(&self) -> (&Id, &State) {
+        (&self.default, self.states.get(&self.default).unwrap())
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct State {
+    pub properties: IndexMap<Name, PropertyVariant>,
+}
+
+#[derive(Debug, Clone)]
+pub enum PropertyVariant {
```

Comment:
Pretty confusing name. I'd go for something like `StatePropertyValue` to make it clear that this is not what comes from the top part of the JSON object

Reply:
Naming is always tough. I'm sickened by the very idea of every time having to specify the name of the struct in which this struct is used as part of its name, which is arguably what `StateProperty` in `StatePropertyValue` is. I originally named it this too, but then verbosity is no longer helpful when everything is verbose.

Reply:
Then at least name it `PropertyValue`